src/App.tsx

```diff
@@ -8,53 +10,171 @@ import { PostsList } from './components/PostsList';
+import { useEffect, useState } from 'react';
+import { User } from './types/User';
+import {
+  addNewComment,
+  deleteComment,
+  getUserPostComments,
+  getUserPosts,
+  getUsers,
+} from './api/post';
+import { Post } from './types/Post';
+import { Comment, CommentData } from './types/Comment';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userSelected, setUserSelected] = useState<User | null>(null);
+  const [showPostLoader, setShowPostLoader] = useState<boolean>(false);
+  const [showCommentLoader, setShowCommentLoader] = useState<boolean>(false);
+  const [buttonAddLoading, setButtonAddLoading] = useState(false);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [showSideBar, setShowSideBar] = useState<number | null>(null);
+  const [postComments, setPostComments] = useState<Comment[]>([]);
+  const [errorsPost, setErrorsPost] = useState<boolean>(false);
+  const [errorsComment, setErrorsComment] = useState<boolean>(false);
+  const [showCommentForm, setShowCommentForm] = useState(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    getUsers().then(res => {
+      setUsers(res as User[]);
+    });
+  }, []);
 
-              <Loader />
+  const handleUserPosts = (user: User) => {
+    setShowPostLoader(true);
+    getUserPosts(user.id)
+      .then(res => {
+        setUserPosts(res as Post[]);
+        setErrorsPost(false);
+      })
+      .catch(() => {
+        setErrorsPost(true);
+      })
+      .finally(() => {
+        setShowPostLoader(false);
+      });
+  };
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  const handleShowComment = (currentPost: Post) => {
+    setShowSideBar(currentPost.id);
+    setShowCommentLoader(true);
+
+    setSelectedPost(currentPost);
+
+    getUserPostComments(currentPost.id)
+      .then(res => {
+        setPostComments(res as Comment[]);
+        setErrorsComment(false);
+      })
+      .catch(() => {
+        setErrorsComment(true);
+      })
+      .finally(() => {
+        setShowCommentLoader(false);
+      });
+  };
+
+  const handleDeleteComment = (commentId: number) => {
+    deleteComment(commentId).then(() => {
+      setPostComments(prevPostComments =>
+        prevPostComments.filter((comment: Comment) => comment.id !== commentId),
+      );
+    });
+  };
```

Comment:
In the `handleDeleteComment` function, you are not handling potential errors that might occur during the deleteComment API call. Always handle errors during API calls.